### PR TITLE
Adding Python package: tscv

### PR DESCRIPTION
**Pull request by WenjieZ**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by WenjieZ** [comment]
@conda-forge/help-python

First time contributing. Review needed.

**Comment by WenjieZ** [comment]
@conda-forge-admin, please ping team

File: recipes/tscv/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: db0f7d02f462013832eab786d133e2db0481e25cd6b82617cb53cecb7565ce78
+
+build:
+  number: 0
```
Comment (dopplershift) [inline recipes/tscv/meta.yaml:13]: This looks like it's pure python, so you can do:
```suggestion
  number: 0
  noarch: python
```

File: recipes/tscv/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  build:
+    - {{ compiler('c') }}
```
Comment (dopplershift) [inline recipes/tscv/meta.yaml:18]: Being pure python, there's no need for a C compiler.
```suggestion
```

**Review by dopplershift — CHANGES_REQUESTED** [review]
Generally in good shape, just a couple minor things to make it noarch.

File: recipes/tscv/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+build:
+  number: 0
+  noarch: python
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  host:
+    - python >=3.6
+    - scikit-learn >=0.22
```
Comment (chrisburr) [inline recipes/tscv/meta.yaml:20]: This probably isn't needed in `host`:
```suggestion
```

File: recipes/tscv/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    - pip
+  run:
+    - python >=3.6
+    - scikit-learn >=0.22
+
+test:
+  imports:
+    - tscv
+    - numpy
+    - sklearn
```
Comment (chrisburr) [inline recipes/tscv/meta.yaml:30]: This should only test imports which come from this specific package directly
```suggestion
```

File: recipes/tscv/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  host:
+    - python >=3.6
+    - scikit-learn >=0.22
+    - pip
+  run:
+    - python >=3.6
+    - scikit-learn >=0.22
```
Comment (chrisburr) [inline recipes/tscv/meta.yaml:24]: It looks like numpy is also a direct dependency of `tscv`:
```suggestion
    - scikit-learn >=0.22
    - numpy
```

**Comment by WenjieZ** [comment]
Build error after committing 2/3 of the suggestions: `ModuleNotFoundError: No module named 'numpy'`

The dependency relation (both source and test) of my package is
```
numpy -> tscv
numpy -> scikit-learn -> tscv
```
@chrisburr

**Comment by chrisburr** [comment]
@WenjieZ The issue is that the package is importing itself in the `setup.py`: https://github.com/WenjieZ/TSCV/blob/5de57c07133fc7a56e862269556e7802a8c97bac/setup.py#L2

As you maintain upstream it should ideally be fixed there. My personal preference is to use `setuptools_scm` but there are several other options which are listed here: https://packaging.python.org/guides/single-sourcing-package-version/#single-sourcing-the-version

**Pull request merged**